gNOI-4.1: Add OSActivateNoReboot and OSInstallForStandbyRP deviations

mojiiba opened the pull request:
This pull adds two deviations `OSActivateNoReboot` and  `OSInstallForStandbyRP` to accommodate devices that install images automatically on  standby rp and does not require separate eboot to activate the new os.

xw-g commented on feature/gnoi/os/tests/osinstall/osinstall_test.go:
```diff
@@ -87,20 +88,33 @@ func TestOSInstall(t *testing.T) {
-	tc.activateOS(ctx, t, false)
-	if tc.dualSup {
+
+	if *deviations.OSActiavteRequiresReboot {
```
This deviation only applies to duaSup case, right? Which means you probably need the if `if tc.dualSup` condition?

mojiiba replied:
no, it applies for both cases since some vendors do not support activating os without doing reboot.

xw-g commented:
+ @vosipchu, @vcheny

xw-g commented on feature/gnoi/os/tests/osinstall/osinstall_test.go:
```diff
@@ -87,20 +88,33 @@ func TestOSInstall(t *testing.T) {
+	}
+
+	if !*deviations.NoOSInstallForStandbyRP && tc.dualSup {
 		tc.transferOS(ctx, t, true)
```
Do you still need transferOS to backup in the case of `NoOSInstallForStandbyRP`?

mojiiba replied:
no, there is no need to trasnfer

xw-g replied:
That's my point.  You still have `tc.transferOS(ctx, t, true)` in your PR, or I read it wrong?

mojiiba replied:
yes, I have it, but it is guarded by  !*deviations.NoOSInstallForStandbyRP.  This code block will be skipped when NoOSInstallForStandbyRP is set to true.

xw-g replied:
ah, sorry I missed the double negative. Still it's a bit hard to read. Say if a dual SUP chassis requires reboot in activation, then you line#93 will reboot the whole chassis immediately, right? which means your line#98 logic will reach to a offline chassis?  Anyway you can help refactor it to make it more readable?

mojiiba replied:
sure, I will do that

mojiiba replied:
It seems I also missed the double negative regarding the noreboot flag and I may need a few more days to address this.

mojiiba replied:
@xw-g  thanks for the patience, I have addressed your comments.

mojiiba commented:
> Hi @mojiiba, could you please fix the typo: OSActiavteRequiresReboot -> OSActivateRequiresReboot
> 
> Thanks!



> Hi @mojiiba, could you please fix the typo: OSActiavteRequiresReboot -> OSActivateRequiresReboot
> 
> Thanks!



> Hi @mojiiba, could you please fix the typo: OSActiavteRequiresReboot -> OSActivateRequiresReboot
> 
> Thanks!

thanks, have addressed the comment

thesrinath commented:
/gcbrun

thesrinath commented:
/gcbrun
